File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.notification import NotificationType


class NotificationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    type: NotificationType
    title: str
    body: str
    metadata_json: dict | None
    read_at: datetime | None
    created_at: datetime

File: frontend/src/types/notification.ts
export type NotificationType =
  | "NEW_APPOINTMENT"
  | "APPOINTMENT_CONFIRMED"
  | "APPOINTMENT_REJECTED"
  | "APPOINTMENT_CANCELLED"
  | "APPOINTMENT_REMINDER"
  | "NEW_MESSAGE"
  | "PAYMENT_CONFIRMED"
  | "PAYMENT_FAILED"
  | "SUBSCRIPTION_RENEWED";

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  body: string;
  metadata_json: Record<string, unknown> | null;
  read_at: string | null;
  created_at: string;
}
